functional_programming_tests/test/main.cpp

```diff
@@ -166,79 +183,117 @@ std_msgs::msg::UInt8MultiArray createUInt8MultiArrayMessageFromPath(
   }
 
   auto const start = Position{request->start.data[0], request->start.data[1]};
   auto const goal = Position{request->goal.data[0], request->goal.data[1]};
 
   // Generate the path using the path generator function that was input
-  auto const path = path_generater(start, goal, occupancy_map);
-
+  auto const path = path_generator(start, goal, occupancy_map);
   if (!path.has_value()) {
-    return empty_response();
+    return tl::unexpected(error::NO_VALID_PATH);
   }
 
-  auto const response =
-      std::make_shared<example_srvs::srv::GetPath::Response>();
-  response->success.data = path.has_value();
-  response->path =
+  auto response = GetPath::Response{};
+  response.success.data = path.has_value();
+  response.path =
       pathing::utilities::createUInt8MultiArrayMessageFromPath(path.value());
 
   return response;
 }
 
-struct MainObject {
-  MainObject()
-      : set_costmap_wrapper_{[this](const std::shared_ptr<
-                                        example_srvs::srv::SetMap::Request>
-                                        request,
-                                    std::shared_ptr<
-                                        example_srvs::srv::SetMap::Response>
-                                        response) -> void {
-          const auto path = pathing::utilities::parseSetMapRequest(request);
+};  // namespace generate_path
+
+struct PathingManager {
+  /**
+   * @brief Manages occupancy_map and path generation for the pathing node
+   */
+  PathingManager()
+      : set_costmap_wrapper_{[this](auto const request, auto response) -> void {
+          auto const path = pathing::utilities::parseSetMapRequest(request);
           if (path) this->map_ = path.value();
           response->success.data = path.has_value();
         }},
-        generate_path_wrapper_{
-            [this](
-                const std::shared_ptr<example_srvs::srv::GetPath::Request>
-                    request,
-                std::shared_ptr<example_srvs::srv::GetPath::Response> response)
-                -> void {
-              response = generate_path_callback(request, this->map_,
-                                                pathing::generate_global_path);
-            }} {}
-
-  std::function<void(const std::shared_ptr<example_srvs::srv::SetMap::Request>,
-                     std::shared_ptr<example_srvs::srv::SetMap::Response>)>
-      set_costmap_wrapper_;
-
-  std::function<void(const std::shared_ptr<example_srvs::srv::GetPath::Request>,
-                     std::shared_ptr<example_srvs::srv::GetPath::Response>)>
-      generate_path_wrapper_;
-
+        generate_path_wrapper_{[this](auto const request, auto response) {
+          auto const print_error = [](std::string_view error)
+              -> tl::expected<GetPath::Response, std::string> {
+            std::cout << error << "\n";
+            return tl::make_unexpected("");
+          };
+
+          auto const return_empty_response = []([[maybe_unused]] auto const)
```

Comment:
Nice with the [[maybe_unused]]

Reply:
Yeah, i generally prefer statements that the compiler can see (even if it doesn't buy me anything)